Dockerfile

```diff
@@ -1,9 +1,154 @@
+RUN conda update -n root conda \
+    && conda clean -afy
+
+# Install python2.7 and 3.6 environments
+RUN conda create -n py27 python=2.7 \
+    && bash -c "source activate py27 && conda install notebook ipykernel -y && ipython kernel install --user --name py27 --display-name 'Python 2.7'" \
+    && conda clean -afy
+# seems there's sometimesa problem with pyzmq so need to reinstall it...
+RUN conda create -n py36 python=3.6 \
+    && bash -c "source activate py36 && pip uninstall pyzmq -y && pip install pyzmq && conda install notebook ipykernel -y && ipython kernel install --user --name py36 --display-name 'Python 3.6'" \
+    && conda clean -afy \
+    && rm -rf ~/.cache/pip
+
+# TODO check if all necessary seems there are duplicate from jupyter/scipy image
+SHELL ["/bin/bash", "-c"]
+# Add libs for python 2.7 env
+#     inherited from saagie/python:2.7 image
+#     installed via pip only
+#     installed via conda
+COPY requirements_conda2.txt requirements_conda2.txt
+COPY --from=PYTHON2 /requirements.txt ./requirements_python2.txt
+COPY requirements_pip2.txt requirements_pip2.txt
+RUN conda install -n py27 --quiet --yes --file requirements_conda2.txt \
+    && . activate py27 \
+    && python -m pip install --no-cache-dir -r requirements_python2.txt \
+    && python -m pip install --no-cache-dir -r requirements_pip2.txt \
+    && conda deactivate \
+    && conda clean -afy \
+    && rm -rf ~/.cache/pip
+# Add libs for python 3.6 env
+#     inherited from saagie/python:3.6 image
+#     installed via pip only
+#     installed via conda
+COPY requirements_conda3.txt requirements_conda3.txt
+COPY --from=PYTHON3 /requirements.txt ./requirements_python3.txt
+COPY requirements_pip3.txt requirements_pip3.txt
+RUN conda install -n py36 --quiet --yes --file requirements_conda3.txt \
+    # Some installed library (scikit-learn) could not be removed so use --ignore-installed \
+    && sed -n '/scikit-learn/p' requirements_python3.txt >> requirements_python3_ignore-installed.txt \
+    && sed -i '/scikit-learn/d' requirements_python3.txt \
+    && . activate py36 \
+    && python -m pip install --no-cache-dir --ignore-installed -r requirements_python3_ignore-installed.txt \
+    && python -m pip install --no-cache-dir -r requirements_python3.txt \
+    && python -m pip install --no-cache-dir -r requirements_pip3.txt \
+    && conda deactivate \
+    && conda clean -afy \
+    && rm -rf ~/.cache/pip
+################ Kernels / Conda envs / requirements PART ENDS #################
+
+
+########################## CUDA PART BEGIN ##########################
+USER root
+
+ENV PATH="${PATH}:/usr/local/nvidia/bin:/usr/local/cuda/bin"
+ENV LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/usr/local/nvidia/lib:/usr/local/nvidia/lib64"
+# nvidia-container-runtime
+ENV NVIDIA_VISIBLE_DEVICES all
+ENV NVIDIA_DRIVER_CAPABILITIES compute,utility
+ENV NVIDIA_REQUIRE_CUDA "cuda>=10.0 brand=tesla,driver>=384,driver<385 brand=tesla,driver>=410,driver<411"
+
+ENV CUDA_VERSION 10.0.130
```

Comment:
Why not but that would split all cuda thing in to parts ... env declaration at the beginning and libs installation somewhere else at the end of the file... If we want to separate cuda one day in a specific build file it would be easier to keep everything together in the other env having all variables at the beginning simplify the upgrade if needed.

Reply:
OK , so its a logical separation ... sounds good to me